cdap-docs/admin-manual/source/operations/logging.rst

```diff
@@ -1,77 +1,573 @@
+    Kafka, groups them by program or application, buffers and sorts them in memory, and
+    finally persists them to files in HDFS. Each of these files corresponds to one program
+    or application, depending on how the grouping is configured. (This is set by the
+    property :ref:`log.publish.partition.key <logging-monitoring-writing-logs-to-kafka>`,
+    described :ref:`below <logging-monitoring-configuring-the-log-saver-service>`.)
+
+    *Note:* These files are configured to rotate based on time and size; they can be
+    changed using the properties ``log.pipeline.cdap.file.max.size.bytes`` and
+    ``log.pipeline.cdap.file.max.lifetime.ms`` in the :ref:`cdap-site.xml
+    <appendix-cdap-default-logging>` file as described in 
+    :ref:`logging-monitoring-log-pipeline-configuration`.
+
+  - In addition, custom log pipelines can be configured by adding an XML file in a
+    prescribed location. Each pipeline buffers log messages in memory and sorts them based
+    on their timestamp. 
+   
+  - In addition to persisting logs to files, the Log.saver also emits metrics about the
+    number of log messages emitted by each program. These metrics can be retrieved by
+    querying the :ref:`CDAP metrics system <http-restful-api-metrics>`.
+      
+  - For security, the files written out to persistent storage in HDFS have permissions set
+    such that they are accessible only by the ``cdap`` user.
+
+.. figure:: /_images/logging/logging-framework.png
+    :figwidth: 100%
+    :width: 800px
+    :align: center
+
+    **CDAP Logging Framework:** From YARN containers, through Kafka and the Log Saver
+    Service, to HDFS
+
+
+.. _logging-monitoring-custom-log-pipelines:
+
+Custom Log Pipelines
+====================
+For a custom log pipeline, create and configure a Logback XML file,
+configuring loggers, appenders, and properties based on your requirements, and place the
+file at the path specified in the ``cdap-site.xml`` file by the property
+``log.process.pipeline.config.dir`` of the ``cdap-site.xml`` file.
+
+Each custom pipeline requires a unique name, which is used for persisting the data and the
+retrieving of metadata. Properties controlling the pipeline (the
+``log.process.pipeline.*`` properties) are described :ref:`above
+<logging-monitoring-cdap-site-xml-configuration>`.
+
+For every XML file in the ``log.process.pipeline.config.dir`` directory, a separate log
+pipeline is created, providing isolation from other log pipelines. As they are separate
+Kafka consumers and processes, each pipeline is independent of |---| and doesn't affect
+the performance of |---| other log pipelines. Though CDAP has been tested with multiple
+log pipelines and appenders, the fewer of each that are specified will provide better
+performance.
+
+.. _logging-monitoring-configuring-custom-log-pipelines:
+
+Configuring Custom Log Pipelines
+--------------------------------
+CDAP looks for Logback XML files located in a directory as set by the property
```

Comment:
is it a "Logback XML file" or a "'logback' file"? Perhaps you can use the same term everywhere. (I don't have preference for either one).